coordinator/src/routes.rs

```diff
@@ -159,16 +164,49 @@ pub async fn index() -> impl IntoResponse {
 }
 
 pub async fn prepare_interceptable_payment(
-    target_node: Path<String>,
     State(app_state): State<Arc<AppState>>,
+    params: Json<PrepareInterceptablePaymentParams>,
 ) -> Result<Json<RouteHintHop>, AppError> {
-    let target_node = target_node.0;
+    let target_node = params.target_node.clone();
     let target_node: PublicKey = target_node.parse().map_err(|e| {
         AppError::BadRequest(format!(
             "Provided public key {target_node} was not valid: {e:#}"
         ))
     })?;
 
+    if let Some(amount_sats) = params.amount_sats {
+        let mut conn = app_state.pool.get().map_err(|e| {
+            AppError::InternalServerError(format!("Could not get connection: {e:#}"))
+        })?;
+
+        let balance = app_state
+            .node
+            .inner
+            .get_on_chain_balance()
+            .map_err(|e| AppError::InternalServerError(format!("Could not get balance: {e:#}")))?
+            .get_spendable();
+
+        let have_enough_liquidity = balance >= amount_sats + MIN_LIQUIDITY_THRESHOLD;
+
+        LiquidityRequest::insert(
```

Comment:
🔧 We should only insert a liquidity request and check the balance if the user intends to open a channel (onboarding). This will be introduced with https://github.com/get10101/10101/pull/1384. Please adapt once this has landed.

Reply:
In hindsight, this ticket should have only be started after #1384 got merged. Rebase was pretty painful :/